### DISCUSSION TITLE
https connection

### DISCUSSION THREAD

**Discussion by yongra** [tl-its-umich-edu/my-learning-analytics#1309] [Q&A]
I think I am missing a step somewhere. I have followed the Readme file to set up the server. I am trying to create the developer key on the Canvas side. When I try the configuration URL, it is not connecting. I think I am missing a step on forwarding HTTPS traffic to the MyLA web server in the docker container. How do set that up? Is this the ngrok install part?

**Comment by jonespm** [comment]
Yeah, because Canvas runs in SSL the LTI applications also need to be SSL (https). The easiest way to serve up from localhost is to use a service like ngrok that gives you the https proxy. 

Otherwise you'd have to either get a DNS name for your local machine, get an SSL cert and configure a proxy server or Gunicorn to serve in https. 

At Michigan when we need to test LTI, we just do local development with ngrok generally just using the free version though we have a few licenses. Though not everything needs to be tested and launched via LTI. In production we are running this behind an HAProxy router that does the SSL termination. Other similar services to ngrok exist like [PageKite](https://pagekite.net/). They also work the same. 

With some effort [you could create your own localhost tunneling service.](https://avenkat-iot.blogspot.com/2019/10/setup-your-own-localhost-tunneling.html) but that's not something I've ever spent time on with the low cost of what's available.

**Reply by yongra** [reply]
thank you. How do you make the SSL connection to the docker container? My server already has DNS. How do I forward the HTTPS traffic? When I try it, HTTP or HTTPS has no connection. Only localhost:5001 exist.

**Reply by yongra** [reply]
If you use HAProxy, you forward the traffic to the MyLA server port 5001?

Also, it is possible to set up apache with WSGI module to make it work as well?

**Reply by jonespm** [reply]
MyLA in the container by default runs on port 5000, so we forward port 5000.

5001 is the port remapped by docker-compose.yaml locally

    ports:
      - "5001:5000"
I'm not sure why we did that, I think it was because we had other apps running on port 5000 at the time. This 5000 can be changed by setting `GUNICORN_PORT` in the environment if needed.
